### PR TITLE
Add CommCare Analytics access levels on HQ

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

For domains where CommCare Analytics feature flag (`SUPERSET_ANALYTICS`) is enabled,
This change lets a domain admin set access (View/Edit) for CommCare Analytics for users and set which CommCare Analytics roles they should have assigned by setting them on the User Roles on CommCareHQ.
This way users get access and roles on CommCare Analytics when they login as set on the User Role assigned to them on CommCareHQ.
Additionally,
For users that have access to view roles on HQ, would be able to see what it set but won't be able to change it. This is again only if the feature flag (`SUPERSET_ANALYTICS`) is enabled.

On UI, we now have two new options available when setting permissions on roles
1. Firstly, set View/Edit for commcare analytics
Screenshot from 2024-04-17 18-29-18
-------
2. Secondly, a new section to set roles (all or specific ones) for commcare analytics
Screenshot from 2024-04-17 18-29-39

## Technical Summary

https://dimagi.atlassian.net/browse/SC-3472
Simply added new permissions for commcare analytics for view, edit and for roles (one for all and one for list of selected ones).
The migration added simply creates the SQL equivalent for the couch model attributes where new permissions are added.

## Feature Flag

`SUPERSET_ANALYTICS`

### Safety story

Tested locally.
The changes that are outside feature flag are safe and should not impact roles and permissions feature otherwise.

### QA Plan

https://dimagi.atlassian.net/browse/QA-6385

### Migrations

- [X] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -224,6 +224,12 @@ class HqPermissions(DocumentSchema):
 
     manage_domain_alerts = BooleanProperty(default=False)
 
+    view_commcare_analytics = BooleanProperty(default=False)
+    edit_commcare_analytics = BooleanProperty(default=False)
+
+    commcare_analytics_roles = BooleanProperty(default=False)
+    commcare_analytics_roles_list = StringListProperty(default=[])
```
Comment (mkangia) [inline corehq/apps/users/models.py:232]: note to self:
Add migration for these once staging is unblocked on conflicts with multiple migrations in parallel to users app.

Comment (Charl1996) [reply corehq/apps/users/models.py:232]: In the `Technical Summary` it says 
> The migration added simply creates the SQL equivalent for the couch model attributes where new permissions are added.

I presume this is migration you're referring to above?

Comment (mkangia) [reply corehq/apps/users/models.py:232]: Hey @Charl1996 
Yes. I had the migration added but then removed it in effort to resolve the staging conflict but there were more conflicts, so I will probably just add it back anyway. Just waiting on some more information on the other PR the conflict was with.
The way the migration is, the code can still be deployed without it and the staging environment won't break.

**Comment by mkangia** [comment]
QA passed.
Updated the branch with master and added the migration that was temporarily removed to unblock staging deploy.

**Comment by mkangia** [comment]
Hey @Charl1996 @ajeety4 

This is now ready for review. QA passed so its go to go.

**Review by Charl1996 — APPROVED** [review]
Looks good to me

**Review by ajeety4 — APPROVED** [review]
Great feature. 
I think we can probably use this later to add the cca role `dataset_editor` which was requested a few times.

**Comment by mkangia** [comment]
Oh thanks for sharing that @ajeety4 

Can you share a ticket where this came up? I will check with team if we want to include it.

**Comment by ajeety4** [comment]
> Oh thanks for sharing that @ajeety4
> 
> Can you share a ticket where this came up? I will check with team if we want to include it.

[This](https://dimagi.atlassian.net/browse/SC-3610) was the recent ticket where this request came up. Prior to this, we had two more requests for this role.

**Pull request merged**